src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediatePreparedStatement.java

```diff
@@ -78,34 +56,25 @@ private int parseType(byte[][] parameters, int index) throws IllegalArgumentExce
     if (this.parameterDataTypes.size() > index) {
       return this.parameterDataTypes.get(index);
     } else {
-      throw new IllegalArgumentException("PgAdapter does not support untyped parameters.");
+      return Oid.UNSPECIFIED;
     }
   }
 
   public List<Integer> getParameterDataTypes() {
     return this.parameterDataTypes;
   }
 
-  public void setParameterDataTypes(List<Integer> parameterDataTypes)
-      throws IllegalArgumentException {
+  public void setParameterDataTypes(List<Integer> parameterDataTypes) {
     this.parameterDataTypes = parameterDataTypes;
-    if (this.parameterDataTypes.size() != this.parameterCount) {
```

Comment:
Check removed as we need to support untyped parameters, which means that there could be more parameters than types.